### ISSUE TITLE
debugGoWithArgumentString() does not properly handle quoted arguments

### ISSUE THREAD

**Issue by bryanforbes** [jakeheis/SwiftCLI#10]
Given the following command:

```swift
CLI.registerChainableCommand(commandName: "greet")
    .withShortDescription("Greets the given person")
    .withSignature("<person>")
    .withExecutionBlock { (arguments) in 
        let person = arguments.requiredArgument("person");
        print("Hey there, \(person)!");
    }
;
```

When using `CLI.go()`, passing quoted arguments properly parses them as single arguments. Here is the result of using it in an app and passing a quoted string:

```sh
$ myApp greet "Bryan Forbes"
Hey there, Bryan Forbes!
```

However, when using `CLI.debugGoWithArgumentSring("myApp greet \"First Name, Last Name\"")`, an error is raised:

```sh
[Debug Mode]
Expected 1 argument, but got 2.
```

The reason is that `CLI.debugGoWithArgumentSring()` splits on spaces without considering quotes which means `"Bryan Forbes"` ends up as two arguments where one is `"Bryan` and the other is `Forbes"`.

**Comment by jakeheis** [comment]
Thanks for the bug report, fixed by efa5105

**Comment by bryanforbes** [comment]
Thank you! This works great.
